aten/src/ATen/native/cuda/PersistentSoftmax.cuh

```diff
@@ -167,6 +167,11 @@ __global__ void softmax_warp_forward(output_t *dst, const input_t *src, int batc
                         elements[i][it] = std::exp(elements[i][it] - max_value[i]);
                         sum[i] += elements[i][it];
                     }
+                } else {
```

Comment:
do we need this else? `elements` (masked and unmasked) are correctly initialized, so previous branch in line 163 should handle them correctly (as long as max_value is finite), we should just remove `if (!mask[idx])`

Reply:
oh I see, we're only "saving" the addition of 0 to the sum and that's not a savings on a GPU because the branch is expensive. I'll give that a shot

Reply:
This causes the not-masked-everywhere tests to fail due to 100% mismatch rate.

Reply:
Would probably still need a branch protecting the addition to sum[i], at which point there's no point, right?

Reply:
> elements (masked and unmasked) are correctly initialized

they're not necessarily initialized to -infinity if they correspond to a masked index, so I think that's why this didn't work.

> Would probably still need a branch protecting the addition to sum[i], at which point there's no point, right?

That didn't help either, probably because the initialization is not magically to -infinity for masked locations.

Reply:
Hm, I thought they are initialized to -inf https://github.com/pytorch/pytorch/pull/75493/files#diff-1dc91d02c649bc32c97da8214879db0fbbd8ed3e342ba7cb6e7de9bb65b79b47R111, but apparently something is not working.